src/lib/installation/update_repositories_finder.rb

```
@@ -124,7 +124,13 @@ def update_url_from_control
     #
     # @return [Array<URI>,false] self-update URLs or false in case of error
     def update_urls_from_connect
-      url = registration_url
+      begin
+        url = registration_url
+      rescue URI::InvalidURIError => e
+        log.error("Custom registration url is wrong, URI failed with: #{e.message}")
```

After discussing with @teclator I think that @lslezak is right and we should report the error. But instead of using `Report` there, I would raise an exception (maybe a custom one) which could be handled in the client to not pollute this class with user interaction stuff.